src/tactic/compute_degree.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.degree.lemmas
+
+/-! # `compute_degree_le` a tactic for computing degrees of polynomials
+
+This file defines the tactic `compute_degree_le`.
+
+Using `compute_degree_le` when the goal is of the form `f.nat_degree ≤ d`, tries to solve the goal.
+It may leave side-goals, in case it is not entirely successful.
+
+See the doc-string for more details.
+
+##  Future work
+
+* Deal with goals of the form `f.(nat_)degree = d` (PR #14040 does exactly this).
+* Add better functionality to deal with exponents that are not necessarily closed natural numbers.
+* Add support for proving goals of the from `f.(nat_)degree ≠ 0`.
+* Make sure that `degree` and `nat_degree` are equally supported.
+
+##  Implementation details
+
+We start with a goal of the form `f.(nat_)degree ≤ d`.  Recurse into `f` breaking apart sums,
+products and powers.  Take care of numerals, `C a, X (^ n), monomial a n` separately. -/
+
+namespace polynomial
+variables {R : Type*}
+
+lemma nat_degree_sub_le_iff_left [ring R] {n : ℕ} (p q : polynomial R) (qn : q.nat_degree ≤ n) :
+  (p - q).nat_degree ≤ n ↔ p.nat_degree ≤ n :=
+begin
+  rw [sub_eq_add_neg, nat_degree_add_le_iff_left],
+  rwa nat_degree_neg,
+end
+
+variables [semiring R] (a : polynomial R)
+
+lemma nat_degree_bit0 : (bit0 a).nat_degree ≤ a.nat_degree :=
+(nat_degree_add_le _ _).trans (max_self _).le
+
+lemma nat_degree_bit1 : (bit1 a).nat_degree ≤ a.nat_degree :=
+(nat_degree_add_le _ _).trans (by simp [nat_degree_bit0])
+
+end polynomial
```

Comment:
Could you move these to a non-tactic file, like `polynomial/degree.lean`?

Reply:
I created a separate PR that only contains these three lemmas: #15522.  I placed them in the same folder that was already imported by this tactic, to avoid increasing the import tree below `compute_degree_le`.